src/Psalm/Internal/Analyzer/Statements/Block/LoopAnalyzer.php

```diff
@@ -722,4 +716,60 @@ private static function getAssignmentMapDepth(string $first_var_id, array $assig
 
         return $max_depth;
     }
+
+    /**
+     * @param PhpParser\Node\Expr[] $post_expressions
+     */
+    private static function analyzePostExpressions(
+        StatementsAnalyzer $statements_analyzer,
+        array $post_expressions,
+        Context $context
+    ): bool {
+        foreach ($post_expressions as $post_expression) {
+            $fake_assignment = null;
```

Comment:
This is a bit weird.

I had to replace the expression in the third argument of for() because otherwise Psalm would do arithmetics on the variable and completely break the assertion from the second argument.